test/functional/wallet_miniscript_decaying_multisig_descriptor_psbt.py

```diff
@@ -0,0 +1,176 @@
+            assert_approx(participant.getbalance(), deposit_amount, vspan=0.001)
+
+        self.log.info("Send a transaction from the multisig! We can always spend with all signers!")
+        to = participants["signers"][self.N - 1].getnewaddress()
+        value = 1.5
+        self.log.info("First, make a sending transaction, created using `walletcreatefundedpsbt` (anyone can initiate this)...")
+        psbt = participants["multisigs"][0].walletcreatefundedpsbt(inputs=[], outputs={to: value}, feeRate=0.00010)
+
+        psbts = []
+        self.log.info("Now at least M users check the psbt with decodepsbt and (if OK) signs it with walletprocesspsbt...")
+        for m in range(self.M):
+            signers_multisig = participants["multisigs"][m]
+            self._check_psbt(psbt["psbt"], to, value, signers_multisig)
+            signing_wallet = participants["signers"][m]
+            partially_signed_psbt = signing_wallet.walletprocesspsbt(psbt["psbt"])
+            psbts.append(partially_signed_psbt["psbt"])
+
+        self.log.info("Finally, collect the signed PSBTs with combinepsbt, finalizepsbt, then broadcast the resulting transaction...")
+        combined = coordinator_wallet.combinepsbt(psbts)
+        finalized = coordinator_wallet.finalizepsbt(combined)
+        coordinator_wallet.sendrawtransaction(finalized["hex"])
+
+        self.log.info("Check that balances are correct after the transaction has been included in a block.")
+        self.generate(self.nodes[0], 1)
+        assert_approx(participants["multisigs"][0].getbalance(), deposit_amount - value, vspan=0.001)
+        assert_equal(participants["signers"][self.N - 1].getbalance(), value)
+
+        self.log.info("Send more transactions from the multisig as required signers decay, this time with a daisy chained signing flow (one after another in series)!")
+        sent = value
+        for locktime in self.locktimes:
+            self.M -= 1
+            self.log.info(f"At block height >= {locktime} this multisig is {self.M}-of-{self.N}")
+            current_height = self.nodes[0].getblock(self.nodes[0].getbestblockhash())['height']
+
+            psbt = participants["multisigs"][0].walletcreatefundedpsbt(inputs=[], outputs={to: value}, feeRate=0.00010, locktime=locktime)
+            for m in range(self.M):
+                signers_multisig = participants["multisigs"][m]
+                self._check_psbt(psbt["psbt"], to, value, signers_multisig)
+                signing_wallet = participants["signers"][m]
+                psbt = signing_wallet.walletprocesspsbt(psbt["psbt"])
+                assert_equal(psbt["complete"], m == self.M - 1)
+            self.log.info("Check that the time-locked transaction is too immature to spend...")
+            assert_equal(current_height >= locktime, False)
+            assert_raises_rpc_error(-26, "non-final", coordinator_wallet.sendrawtransaction, psbt["hex"])
+
+            self.generate(self.nodes[0], locktime)
```

Comment:
In de93b863583921929bc16feb7a0bf5decfbc7ac5 "tests: add functional test for miniscript decaying multisig"

This massively overshoots the locktime block each time. I think it would be better to mine just enough blocks to reach the locktime height.